File: Python/Algorithms/1281.py
class Solution:
    def subtractProductAndSum(self, n: int) -> int:
        s = 0
        p = 1
        while n:
            # extract digits; divmod better than 
            # explicit // and %, C implemented.
            n, d = divmod(n, 10)
            s += d
            p *= d
            
        return p - s

File: Rust/Algorithms/1281.rs
impl Solution {
    pub fn subtract_product_and_sum(n: i32) -> i32 {
        let (mut p, mut s, mut mn) = (1, 0, n);
        // divmod mn and add/multiply values.
        while (mn != 0) {
            let d = mn.rem_euclid(10);
            s += d;
            p *= d;
            mn /= 10;
        }
        p - s
    }
}
